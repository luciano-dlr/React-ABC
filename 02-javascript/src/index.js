const activo = true;

// let mensaje = '';

// if (!mensaje) {

//     mensaje = 'activo'

// } else {

//     mensaje = 'Inactivo'

// }

// const mensaje2 = (activo) ? 'activo' : 'Inactivo';


// console.log(mensaje, mensaje2)

const mensaje = !activo && 'Activo'
const mensaje3 = false && 'Activadisimo'

console.log(mensaje3)
