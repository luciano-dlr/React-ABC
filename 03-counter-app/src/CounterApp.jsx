import { useState } from "react";
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value)

    // c es el valor acutal del counter, recibido por parametro en la funcion anonima dentro del setCounter
    // con retorno inplicito del valor actual de counter 'c' mas '1'
    const handleAdd = () => setCounter((c) => c + 1)

    const handleSubstract = () => setCounter((c) => c - 1);

    const handleReset = () => setCounter(value);


    return (
        <>
            <h1>Counter App</h1>
            <h2>{counter}</h2>
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleSubstract}> -1 </button>
            <button onClick={handleReset}>reset</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number,
}

CounterApp.defaultProps = {
    value: 'Numero no Ingresado',
}

