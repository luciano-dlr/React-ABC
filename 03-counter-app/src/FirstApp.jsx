
export const FirstApp = ({title,subtitle}) => {

   

  return (
    <>
    <h1>{ title }</h1>
    <div>{subtitle }</div>
    </>
  )
}
