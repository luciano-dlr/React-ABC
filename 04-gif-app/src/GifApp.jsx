import { useState } from "react"
import { AddCategory } from "./components/AddCategory";


 export const GifApp = () => {
    
    const [categories,setCategories] = useState(['One Punch','Dragon Ball','leonardo']);

    
    const apiKey = 'LDUkHhGns7VGICYhK8IAuJ8RaonsQk5C'

    return (
        <>
            <h1>Gif App</h1>

            <AddCategory onAddCategories={setCategories}/>
            
            
            <ol>
                {categories.map(category => {
                    return <li key={category}>{category}</li>
                })}
            </ol>
        </>
    )
}


