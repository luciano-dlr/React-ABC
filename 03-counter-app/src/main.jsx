import React from 'react'
import ReactDOM from 'react-dom/client'
import { App } from './HellowWorldApp'
import { FirstApp } from "./FirstApp"
import './styles.css'
import { CounterApp } from './CounterApp'

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <App/>
        {/* <FirstApp title={'Soy un titulo'}  /> */}
        <CounterApp value={5}/>
    </React.StrictMode>
)