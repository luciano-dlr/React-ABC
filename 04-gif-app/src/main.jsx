import './styles.css';
import React from 'react';
import { GifApp } from './GifApp.jsx';
import ReactDOM from 'react-dom/client';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifApp />
  </React.StrictMode>,
)
