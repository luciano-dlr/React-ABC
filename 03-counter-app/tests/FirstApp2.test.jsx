import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en FirstApp', () => {

    const title = 'Hola soy leio';
    const subtitle = 'Soy un subtitle'

    test('debe hacer match con snapshotfirst', () => {

        const { container } = render(<FirstApp tittle={title} />)

        expect(container).toMatchSnapshot();
    })

    test('debe de mostrar el mensaje hola soy leio', () => {


        render(<FirstApp title={title} />)
        expect(screen.getAllByText(title)).toBeTruthy();
        // screen.debug()

    })

    test('debe mostrar el titulo en un h1', () => {

        render(<FirstApp title={title} />)
        expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(title)

    })

    test('debe mostrar subtitule por props', () => {

        render(<FirstApp
            title={title}
            subtitle={subtitle}
        />
        );

        expect(screen.getAllByText(subtitle).length).toBe(1)



    })



})
