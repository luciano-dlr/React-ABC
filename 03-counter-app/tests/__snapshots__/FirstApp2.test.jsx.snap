// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp debe hacer match con snapshotfirst 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    No hay Titulo
  </h1>
  <div>
    No hay subtitulo
  </div>
  <div>
    No hay un nombre
  </div>
</div>
`;
